### ISSUE TITLE
Some bug reports and suggestions

### ISSUE THREAD

**Issue by tshmak** [DigitalPhonetics/IMS-Toucan#110]
1. Rather than continuing step_counter when `not finetune`, suggests continuing when `resume`, e.g.: [here](https://github.com/DigitalPhonetics/IMS-Toucan/blob/3f5a430030d08b82a8e748a0f9252db8a20fc9a9/TrainingInterfaces/Text_to_Spectrogram/PortaSpeech/portaspeech_train_loop.py#L101)
2. Suggest treating `step` differently in `wandb.log` --- `step` has special meaning in wandb.log. See [here](https://docs.wandb.ai/ref/python/log).
3. Suggest adding option for turning off saving spectrograms in training/finetuning
4. BUG: Pretty sure "or" should be "and" [here](https://github.com/DigitalPhonetics/IMS-Toucan/blob/3f5a430030d08b82a8e748a0f9252db8a20fc9a9/Preprocessing/TextFrontend.py#L168)

**Comment by Flux9665** [comment]
1. resume does not automatically mean that we resume a training run. Originally this was the case, but it changed in the past and now the name is misleading. I will think about renaming it in the future. resume means that we load the checkpoint with the highest step count in its filename in the current model directory. It's just a short form of explicitly writing out the most recent checkpoint in the path_to_checkpoint. We can still finetune from this checkpoint that we get when we set the resume flag, in which case the step counter should start from 0. So this should be intended behavior with a misleading name.
2. Yes, that's why we log "Steps" not "step". This way you can set the x-Axis to "Steps" on the website to see all training runs aligned by their number of updates rather than epochs. Did I log "step" somewhere accidentially? I can't find any case where I log "step".
3. I'm not sure why you would want to disable this, it's the only way of seeing whether a model has collapsed, since the losses don't indicate this well. Visually inspecting the outputs is pretty important when using probabilistic models such as the normalizing flow postnet, because the loss can go down significantly, but the spectrograms turn into nonsense. Is there maybe another way of changing this? What part about the spectrogram logging is the reason for wanting to disable it?
4. I think you're right, that looks like a mistake. Thank you!

**Comment by tshmak** [comment]
Thanks for your reply! Here's to continue the discussions: 
1. Thanks for the clarification. Anyway, the current setup is a bit problematic, cos if I'm finetuning but also continuing from a previous checkpoint, I don't get to resume my step_counter... 
2. I mean why don't we do `wandb.log(log_dict, step=step_counter)` rather than having `step_counter` as one of the data points within `log_dict`? This means that if we continue a run, we don't start from 0 step again... But perhaps this is related to the fact that you use a different `name` (in `wandb.init`) for every run (including the continuation of previous runs). However, my practice is to define `project` and `group` only in `wandb.init`. In this case, runs from the same `group` are automatically grouped together. 
3. It seems to me the codes already run pretty well. So for example, when I finetuned your model on my data, I didn't experience any problems. In general, I don't expect to be modifying your architecture much. Anyway, I guess the bottom line is if the model is already well developed, and the user can expect very little chance of it going awry then they can switch off saving spectrograms. In any case, they would be able to tell if the output audio doesn't sound right.

**Comment by Flux9665** [comment]
Regarding 1: finetuning means starting from scratch, but using a previous checkpoint as initialization. So when finetuning, the step counter always starts at 0. If you are finetuning a model and the run crashes and you want to resume this finetuning run, leave out the --finetune the next time you start it, because after the first checkpoint is made, you don't want to start over, but resume from this checkpoint. Only the init before the very first epoch should be made with the --finetune flag.

Regarding 2: Now I understand what you mean. Good point, I will change this. 

Regarding 3: I think this is an important safeguard, because since the introduction of the postflow we are dealing with a sensitive stochastic model, that can at any point randomly collapse while its loss is still decreasing.
